pyvrp/GeneticAlgorithm.py

```diff
@@ -155,7 +155,10 @@ def _educate(self, individual: Individual):
                     ):
                         self._ls.intensify(individual)
 
-                    if individual.cost() < self._best.cost():
+                    if (
+                        individual.is_feasible()
+                        and individual.cost() < self._best.cost()
+                    ):
                         self._best = individual
```

Comment:
It's not needed here because this already happens inside `if individual.is_feasible()`

Reply:
This might still be needed because an intensified solution can become infeasible I think.

Reply:
Yeah that's fair. But then the register calls below this are also wrong. Can you update those?